sys/arm64/arm64/uma_machdep.c

```diff
@@ -56,12 +56,11 @@ uma_small_alloc(uma_zone_t zone, vm_size_t bytes, int domain, u_int8_t *flags,
 	pa = m->phys_addr;
 	if ((wait & M_NODUMP) == 0)
 		dump_add_page(pa);
-	va = (void *)PHYS_TO_DMAP(pa);
-#ifdef __CHERI_PURE_CAPABILITY__
-	va = cheri_setbounds(va, bytes);
-	va = cheri_andperm(va, CHERI_PERMS_KERNEL_DATA |
+	KASSERT(bytes == PAGE_SIZE, ("%s: invalid allocation size %zu",
+	    __func__, bytes));
+	va = (void *)cheri_kern_setbounds(PHYS_TO_DMAP(pa), PAGE_SIZE);
+	va = cheri_kern_andperm(va, CHERI_PERMS_KERNEL_DATA |
```

Comment:
Only morello has this andperm operation.  I'm not sure if it is really warranted (do we have permissions other than CODE | DATA for the direct map?), but if it is it is probably needed on riscv and mips64 as well?

Reply:
Not needed:

https://github.com/CTSRD-CHERI/cheribsd/blob/d3d85fefa5be117458bc629411e4a5f8acb8b23a/sys/arm64/arm64/pmap.c#L905-L906